modules/core/src/test/java/org/apache/ignite/internal/processors/cache/transactions/AtomicOperationsInTxTest.java

```diff
@@ -58,122 +61,186 @@ public class AtomicOperationsInTxTest extends GridCommonAbstractTest {
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.getAndRemoveAsync(1));
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.getAndRemoveAsync(1));
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.getAndReplace(1, 2));
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.getAndReplace(1, 2));
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.getAndReplaceAsync(1, 2).get());
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.getAndReplaceAsync(1, 2).get());
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.remove(1, 1));
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.remove(1, 1));
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.removeAsync(1, 1).get());
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.removeAsync(1, 1).get());
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.removeAll(map.keySet()));
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.removeAll(map.keySet()));
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.removeAllAsync(map.keySet()).get());
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.removeAllAsync(map.keySet()).get());
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.containsKey(1));
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.containsKey(1));
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.containsKeyAsync(1).get());
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.containsKeyAsync(1).get());
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.containsKeys(map.keySet()));
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.containsKeys(map.keySet()));
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.containsKeysAsync(map.keySet()).get());
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.containsKeysAsync(map.keySet()).get());
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.invoke(1, new SetEntryProcessor()));
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.invoke(1, new SetEntryProcessor()));
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.invokeAsync(1, new SetEntryProcessor()).get());
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.invokeAsync(1, new SetEntryProcessor()).get());
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.invokeAll(map.keySet(), new SetEntryProcessor()));
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.invokeAll(map.keySet(), new SetEntryProcessor()));
 
-        checkOperation(isAtomicCacheAllowedInTx, cache -> cache.invokeAllAsync(map.keySet(),
+        checkOperation(withAllowAtomicOpsInTx, cache -> cache.invokeAllAsync(map.keySet(),
             new SetEntryProcessor()).get());
-
-        checkLock(isAtomicCacheAllowedInTx);
```

Comment:
Why did you remove this line?

Reply:
I moved it to checkOperation() and added err as a parameter:
checkLock(withAllowAtomicOpsInTx, err) to enable passing the test below:
`@Test
@WithSystemProperty(key = IGNITE_ALLOW_ATOMIC_OPS_IN_TX, value = "true")
public void testSetOfAtomicOperationsWithinTransactionsCheckFalseSystemPropertyTrue() throws Exception {
        checkOperations(false);
   }`

As a result, it became possible to write the following:
`else if (!withAllowAtomicOpsInTx && err == null) {
            cache = grid(0).cache(DEFAULT_CACHE_NAME);
            eCls = CacheException.class;
            eMsg = "Explicit lock can't be acquired within a transaction.";
        }`

Reply:
Rolled back changes